record_imu_cam/plugin.cpp

```diff
@@ -0,0 +1,99 @@
+#include "common/phonebook.hpp"
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+
+#include <fstream>
+#include <filesystem>
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+
+using namespace ILLIXR;
+
+class record_imu_cam : public plugin {
+public:
+	record_imu_cam(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+	{
+		sb->schedule<imu_cam_type>(id, "imu_cam", [this](switchboard::ptr<const imu_cam_type> datum, std::size_t){
+			this->dump_data(datum);
+		});
+
+		std::string record_data = get_record_data_path();
+		//check folder exist, if exist delete it'
+		system(("rm  -r " + record_data).c_str());
```

Comment:
Prefer to use `boost::filesystem` instead of `system(...)`; for example, what if there is a space in the directory name? Boost will handle that correctly "by default." I think we use `boost::filesystem` somewhere else.